src/trivia/util.h

```diff
@@ -40,6 +40,7 @@
 #include <inttypes.h>
 #include <assert.h>
 #include <stdbool.h>
+#include <stddef.h>
```

Comment:
This is a tricky one. Clang's `stddef.h` provides its own `offsetof`

```c
#if !defined(offsetof) ||                                                      \
    (__has_feature(modules) && !__building_module(_Builtin_stddef))
#define offsetof(t, d) __builtin_offsetof(t, d)
#endif
```

which is a true constexpr unlike `((size_t) &((type *)0)->member)`.

This is what clang 18 has to say about luajit's asserts:

```
tarantool/third_party/luajit/src/lj_obj.h:488:18: note: cast that performs the conversions of a reinterpret_cast is not allowed in a constant expression
  488 | LJ_STATIC_ASSERT(offsetof(Node, val) == 0);
      | ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
```